spec/features/auth/user_lockout_spec.rb

```diff
@@ -0,0 +1,47 @@
+require "rails_helper"
+
+RSpec.describe "User Lockout" do
+  let(:user) { FactoryBot.create(:user) }
+  let(:admin) { FactoryBot.create(:admin_user) }
+  let(:attempt_number) { Devise.maximum_attempts }
+
+  context "when login-in with the wrong user password up to a maximum number of attempts" do
+    before do
+      visit("/users/sign-in")
+      attempt_number.times do
+        fill_in("user[email]", with: user.email)
+        fill_in("user[password]", with: "wrong_password")
+        click_button("Sign in")
+      end
+    end
+
+    it "locks the user account" do
+      visit("/users/sign-in")
+      fill_in("user[email]", with: user.email)
+      fill_in("user[password]", with: user.password)
+      click_button("Sign in")
+      expect(page).to have_http_status(:unprocessable_entity)
+      expect(page).to have_content("Your account is locked.")
+    end
+  end
+
+  context "when login-in with the wrong admin password up to a maximum number of attempts" do
+    before do
+      visit("/admin/sign-in")
+      attempt_number.times do
+        fill_in("admin_user[email]", with: admin.email)
+        fill_in("admin_user[password]", with: "wrong_password")
+        click_button("Sign in")
+      end
+    end
+
+    it "locks the admin account" do
```

Comment:
Should probably also have a test for entering the 2FA code incorrectly 5 times?

Reply:
I thought this was already tested already in its own feature but it seems not. Adding an extra test.

Reply:
Entering an incorrect code is, but not locking after 5 incorrect tries